### ISSUE TITLE
Missing method in org.hibernate.reactive.query.sql.spi.ReactiveNamedNativeQueryMemento with Hibernate ORM 6.6.4

### ISSUE THREAD

**Issue by yrodiere** [hibernate/hibernate-reactive#2041]
It would seem that with the recent changes to ORM 6.6 in 6.6.4, Hibernate Reactive is no longer compatible:

See https://github.com/quarkusio/quarkus/pull/45197

```
2024-12-18T23:40:22.4748605Z 2024-12-18 23:40:22,448 ERROR [io.qua.ver.htt.run.QuarkusErrorHandler] (vert.x-eventloop-thread-1) HTTP Request to /tests/reactiveFindNativeQuery failed, error id: 496193e7-f562-4444-a950-6e4341076c25-1: java.lang.AbstractMethodError: Receiver class org.hibernate.reactive.query.sql.spi.ReactiveNamedNativeQueryMemento does not define or inherit an implementation of the resolved method 'abstract java.lang.Class getResultType()' of interface org.hibernate.query.sql.spi.NamedNativeQueryMemento.
```

**Comment by gavinking** [comment]
For reference, relevant ORM change was this one: https://github.com/hibernate/hibernate-orm/commit/f46967702ce9f5f8c79cf982f095115911fe5b13#diff-dcb6fa9ec160d5998246004c0485f5501337c36103996b7a6822da76c3c3e018

**Comment by DavideD** [comment]
Thanks, one of our classes extends the interface `NamedNativeQueryMemento`, and I need to add the new `getResultType()` method.

Not sure why we don't extends `AbstractNamedQueryMemento`.
I'm working on it.

**Comment by DavideD** [comment]
Fixed as part of https://github.com/hibernate/hibernate-reactive/issues/2043

**Comment by gsmet** [comment]
@DavideD from a schedule perspective, it would be nice if we could upgrade ORM to 6.6.4 in Quarkus before January 10th so that it's included in 3.18 so no big rush to get things done before Christmas but let's tackle it early next year if we can.

**Comment by DavideD** [comment]
Thanks, it shouldn't be a problem.

Just to give a bit of context:
this issue is caused by unfortunate timing. The breaking change was included closed to the release of Hibernate ORM, while I was updating the the job that builds with the snapshots. At the same time, there is some work done on the release scripts of Hibernate ORM and Reactive. I'm waiting for these to be merged.
